tests/director/tests/application_api_test.go

```diff
@@ -492,34 +494,133 @@ func TestCreateApplicationWithDuplicatedNamesWithinTenant(t *testing.T) {
+		appInputGQL, err := testctx.Tc.Graphqlizer.ApplicationRegisterInputToGQL(in)
+		require.NoError(t, err)
+		createReq := fixtures.FixRegisterApplicationRequest(appInputGQL)
+		actualApp := graphql.ApplicationExt{}
+		err = testctx.Tc.RunOperation(ctx, dexGraphQLClient, createReq, &actualApp)
+		require.NoError(t, err)
 
-	// WHEN
-	delReq := fixtures.FixUnregisterApplicationRequest(actualApp.ID)
-	saveExample(t, delReq.Query(), "unregister application")
-	err = testctx.Tc.RunOperation(ctx, dexGraphQLClient, delReq, &actualApp)
+		require.NotEmpty(t, actualApp.ID)
 
-	//THEN
-	require.NoError(t, err)
+		// WHEN
+		delReq := fixtures.FixUnregisterApplicationRequest(actualApp.ID)
+		saveExample(t, delReq.Query(), "unregister application")
+		err = testctx.Tc.RunOperation(ctx, dexGraphQLClient, delReq, &actualApp)
+
+		//THEN
+		require.NoError(t, err)
+	})
+
+	t.Run("Error when application is in formation", func(t *testing.T) {
+		//GIVEN
+		ctx := context.Background()
+		tenantID := tenant.TestTenants.GetIDByName(t, "TestDeleteApplicationIfInFormation")
+		expectedErrorMsg := "graphql: The operation is not allowed [reason=System deletion failed: the system is part of a formation - test-scenario]"
+
+		defaultValue := "DEFAULT"
+		scenarios := []string{defaultValue, "test-scenario"}
+
+		jsonSchema := map[string]interface{}{
+			"type":        "array",
+			"minItems":    1,
+			"uniqueItems": true,
+			"items": map[string]interface{}{
+				"type": "string",
+				"enum": scenarios,
+			},
+		}
+		var schema interface{} = jsonSchema
+
+		fixtures.CreateLabelDefinitionWithinTenant(t, ctx, dexGraphQLClient, ScenariosLabel, schema, tenantID)
+
+		applicationInput := fixtures.FixSampleApplicationRegisterInput("first")
+		applicationInput.Labels = &graphql.Labels{ScenariosLabel: scenarios}
+		appInputGQL, err := testctx.Tc.Graphqlizer.ApplicationRegisterInputToGQL(applicationInput)
+		require.NoError(t, err)
+
+		createApplicationReq := fixtures.FixRegisterApplicationRequest(appInputGQL)
+		application := graphql.Application{}
+
+		err = testctx.Tc.RunOperationWithCustomTenant(ctx, dexGraphQLClient, tenantID, createApplicationReq, &application)
+		require.NoError(t, err)
+		require.NotEmpty(t, application.ID)
+
+		defer func() {
+			fixtures.UpdateApplicationScenariosToDefaultState(t, ctx, dexGraphQLClient, tenantID, application.ID)
+			fixtures.UnregisterApplication(t, ctx, dexGraphQLClient, tenantID, application.ID)
+		}()
+
+		//WHEN
+		req := fixtures.FixUnregisterApplicationRequest(application.ID)
+
+		err = testctx.Tc.RunOperationWithCustomTenant(ctx, dexGraphQLClient, tenantID, req, nil)
+
+		//THEN
+		require.EqualError(t, err, expectedErrorMsg)
+	})
+
+	t.Run("Error when application is in formation and runtime", func(t *testing.T) {
+		//GIVEN
+		expectedErrorMsg := "graphql: The operation is not allowed [reason=System deletion failed: the system is part of a formation - test-scenario and of runtime - one-runtime]"
```

Comment:
use the word scenario instead (rename everywhere)

Reply:
I think the error message shall stay `formation` as it is by requirements. However, I'll change the method, variable, etc namings to `scenario`

Reply:
Can remain scenario. Extracting scenario label as formation entity will include adjusting all errors as well. This error will not be propagated to users anyway atm.